ws-client/src/jsonrpc_transport.rs

```diff
@@ -15,81 +20,97 @@ pub async fn websocket_connection(config: WsConfig<'_>) -> Result<(Sender, Recei
 }
 
 /// JSONRPC WebSocket sender.
-/// It's a wrapper over `WebSocket sender` with additional `JSONRPC request_id`.
 #[derive(Debug)]
 pub struct Sender {
-	request_id: u64,
 	transport: transport::Sender,
 }
 
 impl Sender {
 	/// Creates a new JSONRPC sender.
 	pub fn new(transport: transport::Sender) -> Self {
-		Self { transport, request_id: 0 }
+		Self { transport }
 	}
 
-	/// Inner implementation for starting either a request or a subscription.
-	async fn start_impl(
+	/// Sends a request to the server but it doesn’t wait for a response.
+	/// Instead, you have keep the request ID and use the Receiver to get the response.
+	///
+	/// Returns Ok() if the request was successfully sent otherwise Err(_).
+	pub async fn start_request(
```

Comment:
you may wonder why we don't do the same for the receiver but the reason is that we move the receiver into a `stream::unfold` which takes ownership of the stream.

so won't work without moving the `request_manager` to a Mutex which I prefer not to do even though it would make code in the background to be easier to read.